### PR TITLE
Spelling and sentence structure fixes

### PR DESCRIPTION
**Pull request by CuriouslyCurious**
As per request I have gone through and touched up a few rough patches in the page. Changed some sentence structures to what I believe is the original meaning. Please check if I misunderstood stuff in places, as I am not quite familiar with the inner-workings of this project.

Seeing as I still have some questions that may help to clarify certain things I'll keep this PR as a draft. I'll add them as comments to the places in the document if possible. Also, seeing how I can't atm convert .org files to .html in the same way as was done here I could use some assistance. :P

Nitpicking aside, well done @tmplt! Proud of you! <3

### CODE REVIEW

File: index.org [hunk 1]
```diff
@@ -208,20 +208,20 @@ Now, let us list and replay the trace we just recorded:
...
+- Source :: a [trace] **source** is any implementation of [[https://github.com/rtic-scope/cargo-rtic-scope/blob/master/src/sources/mod.rs#L20][=trait Source=]] from which decoded trace packets can be pulled via an iterator using for example [[https://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.next][=Iterator::next=]].
   A source can be a live target via =DAPSource= (e.g. an STLink), =TTYSource= (i.e. a ~/dev/tty*~ device), or a file on disk via =FileSource=.
-- Sink :: a [trace] sink is any implementation of =trait Sink= to which decoded trace packets can be [[https://github.com/rtic-scope/cargo-rtic-scope/blob/master/src/sinks/mod.rs][=Sink::drain=]]ed (alt. "forwarded").
+- Sink :: a [trace] **sink** is any implementation of =trait Sink= to which decoded trace packets can be [[https://github.com/rtic-scope/cargo-rtic-scope/blob/master/src/sinks/mod.rs][=Sink::drain=]]ed (alt. "forwarded").
   A sink can be a file on disk via =FileSink= or a frontend via =FrontendSink=.
 
-~cargo-rtic-scope~  abstracts its operation into a single sink and set of sinks.
+~cargo-rtic-scope~ abstracts its operation into a single sink and set of sinks.
```
Comment (CuriouslyCurious) [inline index.org:216]: Not sure what this means. Does it abstract its operation into a set of sinks or into a single sink or both?

File: index.org [hunk 2]
```diff
@@ -270,7 +271,7 @@ For example, when executing =cargo rtic-scope --bin blinky [options]=:
    For example, =device = stm32f4::stm32f401=, =binds = SysTick=, and =binds = EXTI1= might be extracted.
    Here, each IRQ label is associated with the RTIC task it is bound to.
 
-   This parsing step places some restrictions on how the source code for an RTIC application can be written. Refer to [[RTIC application constrains]].
+   This parsing step places some restrictions on how the source code for an RTIC application can be written. Refer to [[RTIC application constraints]].
 3. An adhoc cdylib crate is then built which translates IRQ labels to IRQ numbers.
```
Comment (CuriouslyCurious) [inline index.org:275]: What is a cdylib crate?

File: index.org [hunk 3]
```diff
@@ -270,7 +271,7 @@ For example, when executing =cargo rtic-scope --bin blinky [options]=:
...
    ~libadhoc~ will look like the following with the parse information from the previous step:
```
Comment (CuriouslyCurious) [inline index.org:276]: Where did *libadhoc* come from? Is that related to the cdylib crate?

File: index.org [hunk 4]
```diff
@@ -299,7 +300,7 @@ This ~(task map, reset timestamp, trace clock frequency)~ tuple constitutes the
 If the input buffer of the serial device is filled, packets will be lost or corrupted.
 A warning will be printed before this buffer is overflowed, or if the buffer size cannot be determined.
 
-** RTIC application constrains
+** RTIC application constraints
```
Comment (CuriouslyCurious) [inline index.org:303]: I don't think it is clear why this is a constraint. What is the desired functionality and why is the macro definition not quite ideal?

File: index.org [hunk 5]
```diff
@@ -317,24 +318,23 @@ See [[https://github.com/rtic-scope/examples/blob/master/src/bin/blinky-noconf.r
```
Comment (CuriouslyCurious) [inline index.org:318]: This applies to the points above: What does consume mean in this case? And effectively how? Is it up to the programmer to create a comparator that can be consumed effectively? It is ambiguous.

File: index.org [hunk 6]
```diff
@@ -358,5 +358,5 @@ Project maintainer can be reached via email at [[mailto:v@tmplt.dev][<v@tmplt.de
 
 * Footnotes
 
-[fn:1] This received payload is
+[fn:1] This received payload is.
```
Comment (CuriouslyCurious) [inline index.org:361]: This footnote is seemingly never used.

File: index.org (hunk 1 above)
Comment (tmplt) [reply index.org:216]: A single source feeds into multiple sources.

Comment (tmplt) [reply index.org:216]: Perhaps "`cargo-rtic-scope` operates by forwarding packets from a single source to multiple sources."? Next sentence will need to be amended.

File: index.org [hunk 7]
```diff
@@ -12,51 +12,51 @@
...
     dummy: @1625485615052769556 ns (+76688 ns): [Task { name: "app::toggle", action: Entered }]
     dummy: @1625485615052790806 ns (+21250 ns): [Task { name: "app::toggle", action: Exited }]
   #+end_src
 
-  At present, information only flows from the backend to the frontend.
-  This is subject to change.
-  With bidirectional communication a frontend can for example implement a complex hardware-in-the-loop testing suite.
-  Please refer to the [[Roadmap][project roadmap]].
-- The target-side tracing crate :: ~cortex-m-rtic-trace~, is an small auxiliary crate applied to the target application under trace.
+  At present, information only flows from the backend to the frontend dummy.
```
Comment (tmplt) [inline index.org:31]: A non-dummy frontend can be specified via `--frontend`. "[...] to the selected frontend" perhaps?

File: index.org [hunk 8]
```diff
@@ -12,51 +12,51 @@
...
+  In the future, bidirectional communication will be possible, enabling a frontend
+  to implement a complex hardware-in-the-loop testing suite.
+  Please refer to the [[Roadmap][project roadmap]] for future prospects.
+- The target-side tracing crate :: ~cortex-m-rtic-trace~, is a small auxiliary crate applied to the target application when tracing.
```
Comment (tmplt) [inline index.org:35]: The crate's effect is more permanent than just during tracing what with it altering source code, but "application under trace" does not flow well here either.

 "[...] applied to the traced application"?

File: index.org [hunk 9]
```diff
@@ -12,51 +12,51 @@
...
   It only exposes the =#[trace]= macro, which is used to trace RTIC software tasks.
-  While hardware tasks are traced "free of charge" (see [[How it works]]), software tasks are traced by writing to a =u32=-variable twice.
+  NOTE: While hardware tasks are traced "free of charge" (see [[How it works]]), software tasks are traced by writing to a =u32=-variable twice.
 
 ** Project repositories/crates
 The framework is managed under the [[https://github.com/rtic-scope][RTIC Scope organization on GitHub]].
-Below is a list of the chief repositories that constitute the project.
+Below is a list of the main repositories that constitute the RTIC project.
```
Comment (tmplt) [inline index.org:41]: RTIC -> RTIC Scope

File: index.org [hunk 10]
```diff
@@ -208,20 +208,20 @@ Now, let us list and replay the trace we just recorded:
 We can read from the ~dummy~ frontend that toggling a LED takes about 21µs in debug mode.
 
 * Concepts
-- Source :: a [trace] source is any implementation of [[https://github.com/rtic-scope/cargo-rtic-scope/blob/master/src/sources/mod.rs#L20][=trait Source=]] from which decoded trace packets can be pulled via =Iterator::next=.
+- Source :: a [trace] **source** is any implementation of [[https://github.com/rtic-scope/cargo-rtic-scope/blob/master/src/sources/mod.rs#L20][=trait Source=]] from which decoded trace packets can be pulled via an iterator using for example [[https://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.next][=Iterator::next=]].
```
Comment (tmplt) [inline index.org:211]: Just single `*` in Org-mode for bold. (Italics are done `/like so/`.)

The sentence drags on a little too long. `next` is the only function that needs to be implemented for the `Iterator` trait, so I think just sauing `Interator::next` suffices.

File: index.org (hunk 2 above)
Comment (tmplt) [reply index.org:275]: Rust for shared objects (.so-files). Probably better to just say "a shared object file".

File: index.org (hunk 3 above)
Comment (tmplt) [reply index.org:276]: *libadhoc* is a cdylib (shared object file). The name is because I didn't really know what else to call it: its only purpose is to be built and loaded during setup and its content depends on the target application; its always built in an adhoc manner, hence the name. I'm open to renaming it.

File: index.org (hunk 4 above)
Comment (tmplt) [reply index.org:303]: What constraint do you refer to?

File: index.org (hunk 5 above)
Comment (tmplt) [reply index.org:318]: The DWT comparator is a hardware resource that the user must assign during `init`. See https://github.com/rtic-scope/examples/blob/master/src/bin/blinky.rs#L49 for example.

File: index.org (hunk 6 above)
Comment (tmplt) [reply index.org:361]: Oops. It isn't, no. It can be removed.

**Review by tmplt — CHANGES_REQUESTED** [review]
Thanks for the proposed changes and comments!

File: index.org (hunk 1 above)
Comment (CuriouslyCurious) [reply index.org:216]: Yeah, something like that would work.
